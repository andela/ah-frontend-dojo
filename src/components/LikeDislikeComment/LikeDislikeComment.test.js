// react libraries
import React from 'react';

// third-party libraries
import { shallow } from 'enzyme';

// components
import { LikeDislikeComment, mapDispatchToProps } from '.';

describe('tests the UI functionality of LikeDislikeComment component', () => {
  const mockedFunction = jest.fn();
  const component = shallow(
    <LikeDislikeComment
      articleSlug="test-article"
      commentId={6}
      likeComment={mockedFunction}
      dislikeComment={mockedFunction}
      getLikeCommentState={mockedFunction}
    />
  );


  it('test if likedislike component renders well', () => {
    expect(component).toMatchSnapshot();
  });

  it('calls likeComment function', () => {
    component.setProps({
      likeComment: mockedFunction,
      dislikeComment: mockedFunction,
      getLikeCommentState: mockedFunction,
    });
    mapDispatchToProps(mockedFunction).likeComment();
    expect(mockedFunction).toHaveBeenCalled();
  });

  it('calls dislikeComment function', () => {
    component.setProps({
      likeComment: mockedFunction,
      dislikeComment: mockedFunction,
      getLikeCommentState: mockedFunction
    });
    mapDispatchToProps(mockedFunction).dislikeComment();
    expect(mockedFunction).toHaveBeenCalled();
  });

  it('submits a like for a specific comment', () => {
    component.setProps({
      likeComment: mockedFunction,
      dislikeComment: mockedFunction,
      getLikeCommentState: mockedFunction,
      articleSlug: 'heooe0-he',
      commentId: 4
    });
    const instance = component.instance();
    instance.onSubmitLikeHandler();
    expect(instance.props.dislikeComment).toHaveBeenCalled();
  });

  it('submits a dislike for a specific comment', () => {
    component.setProps({
      likeComment: mockedFunction,
      dislikeComment: mockedFunction,
      getLikeCommentState: mockedFunction,
      articleSlug: 'heooe0-he',
      commentId: 4
    });
    const instance = component.instance();
    instance.onSubmitDisikeHandler();
    expect(instance.props.dislikeComment).toHaveBeenCalled();
  });

  it('should fetch comment like status', () => {
    component.setProps({
      likeComment: mockedFunction,
      dislikeComment: mockedFunction,
      articleSlug: 'heooe0-he',
      commentId: 4,
      getLikeCommentState: mockedFunction,
    });
    const instance = component.instance();
    instance.onSubmitDisikeHandler();
    expect(instance.props.getLikeCommentState).toHaveBeenCalled();
  });
});
