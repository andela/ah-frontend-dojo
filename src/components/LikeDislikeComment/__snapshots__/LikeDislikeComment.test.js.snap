// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the UI functionality of LikeDislikeComment component test if likedislike component renders well 1`] = `
<LikeDislike
  dislikeFunction={[Function]}
  dislikeState={false}
  likeFunction={[Function]}
  likeState={false}
/>
`;
